### PR TITLE
Lauren finding words

### PR DESCRIPTION
**Pull request by lsunn**

### CODE REVIEW

File: src/Board.js [hunk 1]
```diff
@@ -1,52 +1,169 @@
...
+    };
+
+    const handleMouseUp = (rowIndex, colIndex) => {
+        if (isDragging) {
+            setIsDragging(false);
+            if (startCell.rowIndex === rowIndex && startCell.colIndex === colIndex) {
+                handleCellClick(rowIndex, colIndex);
+            } else {
+                checkWord();
+        }
```
Comment (puzonevan) [inline src/Board.js:90]: boop indentation pls

File: src/Board.js [hunk 2]
```diff
@@ -1,52 +1,169 @@
...
             {grid.map((row, rowIndex) => (
                 <div key={rowIndex} className="row">
                     {row.map((cell, colIndex) => (
-                        <Cell key={colIndex} cell={cell} onclick="chooseLetter()"/>
+                        <Cell key={colIndex}
+                        cell={cell}
+                        onClick={() => handleCellClick(rowIndex, colIndex)}
+                        onMouseDown={() => handleMouseDown(cell, rowIndex, colIndex)}
+                        onMouseEnter={() => handleMouseEnter(rowIndex, colIndex)}
+                        onMouseUp={() => handleMouseUp(rowIndex, colIndex)}/>
```
Comment (puzonevan) [inline src/Board.js:166]: I see what's happening here. Usually I wouldn't mind the logic living here. Logic being handleMouseDown, handleMouseEnter, etc. Buttttt, it shouldn't live here. That logic belongs in the `<Cell />` component

```suggestion
                        cell={cell}
                        position={[rowIndex, colIndex]}
                        setIsDragging={setIsDragging}
                        setChosenLetters={setChosenLetters}
                        setSelectedCells={setSelectedCells}
                        ...
```

There is argument to be made that is more props, but doing that allows the logic `handleMouseDown` etc. to live in the `<Cell />` component overall reducing the bloat on this `<Board />` component.

Remember this board component is the most high level component. Functional Logic that should live in a children should live in the children. The high level component should contain high level logic like the state and `checkWord` can actually live in this component still

**Review by puzonevan — CHANGES_REQUESTED** [review]

**Comment by puzonevan** [comment]
Nice work on this. Let's establish some more process for PR's 

1. Title - no need to put your name. Just make sure it's meaningful "Implement finding words"
2. Set up a description template . Let's start with 3 sections `Overview - What does this ticket accomplish` `Test Url - deploy link for me to look at` `Test Scenarios - step by step scenarios of things to test`

File: package.json [hunk 3]
```diff
@@ -7,6 +7,8 @@
     "@testing-library/jest-dom": "^5.17.0",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "dictionary-en": "^4.0.0",
+    "nspell": "^2.1.5",
```
[the remark this answers is not in the document]
Comment (lsunn) [reply package.json:11]: nope not used yet but I think I might use a different library instead

**Pull request merged**